six/modules/c++/six.sicd/include/six/sicd/CropUtils.h

```diff
@@ -29,11 +29,32 @@
 #include <types/RowCol.h>
 #include <scene/Types.h>
 #include <six/NITFReadControl.h>
+#include <six/sicd/ComplexData.h>
 
 namespace six
 {
 namespace sicd
 {
+/*!
+ * Projects a slant plane pixel into ground plane LLA coordinates
+ */
+class PixelToLatLon
+{
+public:
+    PixelToLatLon(const six::sicd::ComplexData& data,
+                  const scene::SceneGeometry& geom,
+                  const scene::ProjectionModel& projection);
+
+    scene::LatLon operator()(size_t row, size_t col) const;
+
+private:
+    const scene::SceneGeometry& mGeom;
```

Comment:
Note that `mGeom` and `mProjection` are just held by const ref.  This was definitely safe when in the unnamed namespace since we knew exactly how it was being used but now it can be screwed up.  I'd hold `mGeom` by value (copying it is trivially cheap) but you can't copy `mProjection` (it's an abstract base class).  For the time being I'd say still hold it by const ref but make it clear in the constructor comments (the alternative is to pass it in by shared pointer which is safer but will force some rippling API changes... I don't have a defensible argument against doing this though).